SerialPrograms/Source/PokemonSV/Inference/Overworld/PokemonSV_OverworldDetector.cpp

```diff
@@ -124,6 +142,71 @@ bool OverworldDetector::detect_ball(const ImageViewRGB32& screen) const{
     return false;
 }
 
+std::pair<double, double> OverworldDetector::locate_ball(const ImageViewRGB32& screen) const{

```

Comment:
Do we still need the `detect_ball()` function above? Can probably delete that and have the caller call `locate_ball()` instead.

Reply:
`detect_ball` has looser requirements than `locate_ball`. Since `locate_ball` requires a min_object_size of 150, while `detect_ball` has no requirement. For example, this means `detect_ball` will return true even if it only detects the yellow arrow, and not the ball itself.

This is more of a theoretical concern, though. I believe `locate_ball` should be able to reliably detect the overworld. Thoughts?

Reply:
Another factor is that I'm actually thinking of reducing the RMSD threshold for `locate_ball`, down to 40. Since the yellow arrow has an RMSD of 60, while the ball itself usually has an RMSD < 20.

Reply:
IIRC, the only other user of the radar ball is the overworld detector. So that's the case we need to make sure still works. The corner case being if you're standing next to an event den, the sparkling animation messes things up.

I don't remember the exact details of the overworld detector, but it uses some combination of the map not moving and the radar ball to detect if you're in the overworld. One of them is a fallback for the other in case the sparkling den information messes things up.

So as long as the overworld detector still works, we should be good. Though radar ball has two shades of yellow which is somewhat sensitive to capture card distortion. That may be wide the filter list is so long.

Reply:
Yeah. I can do some basic testing to make sure the overworld detector still works (including near sparkling dens). I still worry about weird corner cases (e.g. different capture cards) where the old detection method works (with detect_ball), but the new one doesn't (locate_ball).

Are you ok with this risk? Or should we just leave it as is, with detect using detect_ball instead of locate_ball.

Reply:
Yeah that's fine. No real reason to fork that much code if that's being used for the same exact thing. If it breaks the overworld detector, we should know pretty easily.

Reply:
I decided to add a flag `strict_requirements`. When set to true, it uses a tighter RMSD threshold and has min size requirements for waterfill objects. When set to false, it's more similar to the old `detect_ball` function, where the RMSD threshold is looser and there are no min size requirements.